operator/v1/types_ingress.go

```diff
@@ -282,6 +287,339 @@ var (
+	//     - AES128-GCM-SHA256
+	//     - AES256-GCM-SHA384
+	//     - AES128-SHA256
+	//     - AES256-SHA256
+	//     - AES128-SHA
+	//     - AES256-SHA
+	//     - DES-CBC3-SHA
+	//     - !DSS
+	//   securityProtocol:
+	//     minimumVersion: TLSv1.0
+	//     maximumVersion: TLSv1.2
+	//   dhParamSize: 2048
+	//
+	// +optional
+	// +nullable
+	Intermediate *IntermediateSecurityProfile `json:"intermediate,omitempty"`
+	// modern is a security profile based on:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	//
+	// and looks like this (yaml):
+	//
+	//   ciphers:
+	//     - ECDHE-ECDSA-AES256-GCM-SHA384
+	//     - ECDHE-RSA-AES256-GCM-SHA384
+	//     - ECDHE-ECDSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-CHACHA20-POLY1305
+	//     - ECDHE-ECDSA-AES128-GCM-SHA256
+	//     - ECDHE-RSA-AES128-GCM-SHA256
+	//     - ECDHE-ECDSA-AES256-SHA384
+	//     - ECDHE-RSA-AES256-SHA384
+	//     - ECDHE-ECDSA-AES128-SHA256
+	//     - ECDHE-RSA-AES128-SHA256
+	//   securityProtocol:
+	//     minimumVersion: TLSv1.2
+	//     maximumVersion: TLSv1.2
+	//   dhParamSize: 2048
+	//
+	// +optional
+	// +nullable
+	Modern *ModernSecurityProfile `json:"modern,omitempty"`
+	// custom is a user-defined security profile. An example custom profile
+	// looks like this:
+	//
+	//   ciphers:
+	//     - ECDHE-ECDSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-CHACHA20-POLY1305
+	//     - ECDHE-RSA-AES128-GCM-SHA256
+	//     - ECDHE-ECDSA-AES128-GCM-SHA256
+	//   securityProtocol:
+	//     minimumVersion: TLSv1.1
+	//     maximumVersion: TLSv1.2
+	//   dhParamSize: 1024
+	//
+	// Any unset fields of a custom security profile default to the IntermediateSecurityProfile.
+	// 
+	// +optional
+	// +nullable
+	Custom *CustomSecurityProfile `json:"custom,omitempty"`
+}
+
+type OldSecurityProfile struct{}
+type IntermediateSecurityProfile struct{}
+type ModernSecurityProfile struct{}
+
+// CustomSecurityProfile defines the schema for a custom security profile.
+type CustomSecurityProfile struct {
+	SecurityProfile `json:"securityProfile"`
+}
+
+// SecurityProfileType defines a type of security profile.
+type SecurityProfileType string
+
+const (
+	// Old is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	SecurityProfileOldType SecurityProfileType = "Old"
+	// Intermediate is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	SecurityProfileIntermediateType SecurityProfileType = "Intermediate"
+	// Modern is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	SecurityProfileModernType SecurityProfileType = "Modern"
+	// Custom is a security profile that allows for user-defined parameters.
+	SecurityProfileCustomType SecurityProfileType = "Custom"
+)
+
+// SecurityProfile defines the schema for a security profile.
+type SecurityProfile struct {
```

Comment:
throughout: I'm not sure "SecurityProfileType" refers to anything in the openapi output — might want to refer to it like "security profile type" or something?